extract.py

```diff
@@ -0,0 +1,110 @@
+    parser.add_argument('-c', '--container', help='docker container', default='fkiecad/fact_extractor')
+    parser.add_argument('-m', '--memory', help='memory limit for docker container (in MB)', default='512')
+    parser.add_argument('-o', '--output_directory', help='path to extracted files', default=None)
+    parser.add_argument('-r', '--report_file', help='write report to a file', default=None)
+    parser.add_argument('-V', '--verbose', action='store_true', default=False, help='increase verbosity')
+    parser.add_argument('ARCHIVE', type=str, nargs=1, help='Archive for extraction')
+
+    return parser.parse_args()
+
+
+def setup_logging(verbose):
+    console_log = logging.StreamHandler()
+    console_log.setFormatter(
+        logging.Formatter(fmt='[%(asctime)s][%(module)s][%(levelname)s]: %(message)s', datefmt='%Y-%m-%d %H:%M:%S')
+    )
+
+    logger = logging.getLogger('')
+    logger.setLevel(logging.DEBUG if verbose else logging.INFO)
+    logger.addHandler(console_log)
+
+
+def container_exists(container):
+    return subprocess.run('docker history {}'.format(container), shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE).returncode == 0
+
+
+def call_docker(input_file, container, target, report_file, memory_limit, tmpdir=None):
+    tmpdir = tmpdir if tmpdir else TemporaryDirectory()
+
+    for subpath in ['files', 'reports', 'input']:
+        Path(tmpdir.name, subpath).mkdir(exist_ok=True)
+
+    shutil.copy(input_file, str(Path(tmpdir.name, 'input', Path(input_file).name)))
+
+    subprocess.run('docker run --rm -m {}m -v {}:/tmp/extractor -v /dev:/dev --privileged {}'.format(memory_limit, tmpdir.name, container), shell=True)
+
+    logging.warning('Now taking ownership of the files. You may need to enter your password.')
+
+    subprocess.run('sudo chown -R {} {}'.format(os.environ['USER'], tmpdir.name), shell=True)
+    with suppress(shutil.Error):
+        shutil.copytree(str(Path(tmpdir.name, 'files')), target)
+
+    handle_report(report_file, tmpdir.name)
+
+    tmpdir.cleanup()
+
+
+def handle_report(report_file, tmp):
+    indented_report = json.dumps(json.loads(Path(tmp, 'reports', 'meta.json').read_text()), indent=4)
+    if report_file:
+        Path(report_file).write_text(indented_report)
+    else:
+        print(indented_report)
+
+
+def main():
+    arguments = parse_arguments()
+    setup_logging(arguments.verbose)
+
+    output_directory = arguments.output_directory if arguments.output_directory else str(Path(__file__).parent / 'extracted_files')
+    if Path(output_directory).exists():
+        logging.error('Target directory exists ({}). Please choose a non-existing directory with -o option.'.format(output_directory))
+        return 1
+
+    if not container_exists(arguments.container):
+        logging.error('Container {} doesn\'t exist. Please specify an existing container with the -c option.'.format(arguments.container))
```

Comment:
Should we give a hint to download the "default" container as well. Someone might forget it.